File: 프로그래머스/unrated/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.js
function solution(myString, pat) {
    const arr = [];
    
    for (let i = 0; i < myString.length - pat.length + 1; i++) {
        arr.push(myString.slice(i, pat.length + i));
    }
    
    return arr.filter(el => el === pat).length;
}

File: 프로그래머스/unrated/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public int solution(String myString, String pat) {
        List<String> result = new ArrayList<>();
        
        for (int i = 0; i < myString.length() - pat.length() + 1; i++) {
            result.add(myString.substring(i, pat.length() + i));
        }
        
        return result.stream().filter(sub -> sub.equals(pat)).toArray().length;
    }
}